src/components/TodoForm.tsx

```diff
@@ -0,0 +1,49 @@
+
+import React from "react"
+
+function TodoForm(props: {
+        todos: any[]; onAdd: (arg0: {
+            // add 1 to current largest id
+            id: number; text: string; done: boolean;
+        }) => void;
+    }) {
+    const [formData, setFormData] = React.useState(
+      {
+        // add 1 to current largest id
+        id: Math.max(...props.todos.map(val => val.id)) + 1,
```

Comment:
You're assigning a id here, but then reassigning it in `addTodo` in App.js. 

TodoForm shouldn't care about the id or done state. You could remove most of this code by handling these when you add the todo in `addTodo` in `App.js` and avoid passing the todos to the TodoForm entirely.